pkg/registry/generic/registry/store.go

```diff
@@ -40,6 +41,10 @@ import (
 	"github.com/golang/glog"
 )
 
+// EnableGarbageCollector affects the handling of Update and Delete requests. It
+// must be synced with the corresponding flag in kube-controller-manager.
+var EnableGarbageCollector bool
```

Comment:
Pretty barfy, but I understand why you want to do it. This may not be the best place but I can live with it.